conans/client/rest/file_uploader.py

```diff
@@ -48,7 +48,7 @@ def _dedup(self, url, headers, auth):
 
     def exists(self, url, auth):
         response = self._requester.head(url, verify=self._verify_ssl, auth=auth)
-        return response
+        return bool(response.ok)
```

Comment:
I have tested this change manually against artifactory, to see if it could break the ``backup_sources`` feature, and everything keeps working with it (and it is necessary for the TestingResponse framework)